### PR TITLE
Change information holder to `SubmittedExperimentInfo`

### LINKED ISSUE #103: Make a new information holder for `scheduler.py`
### Feature you want to implement
As mentioned in a [discussion](https://github.com/snu-quiqcl/iquip/pull/89#discussion_r1289545223) in PR #89, I will make an information holder class `SubmittedExperimentInfo` (class name subject to change) that `scheduler.py` uses to hold information of queued experiments.

### How the feature is implemented
This new class in `protocols.py` will make an instance whose attributes are:
- `rid`: The run identifier value of the experiment
- `pipeline`: The pipeline of the experiment
- `expid`: The overall information of the experiment, which is a dictionary that may include `arguments`, `file`, etc.
- `priority`: The priority of the experiment
- `due_date`: The due date of the experiment
- `status`: Current state of the experiment
- etc.

### PR DESCRIPTION
**Pull request by giwon2004**
This changes the information holder of `scheduler.py` from `ExperimentInfo` to `SubmittedExperimentInfo` which holds different attributes of the experiment.

For example, if I add a code
```
self.addExperiment(SubmittedExperimentInfo(
                       rid = 1,
                       priority = 1,
                       status = "preparing",
                       pipeline = "main",
                       expid = {"file": "DIR_PATH"},
                       due_date = "2023-08-13"
                   ))
```
Then the scheduler gives
image

This affects #89, #102, and especially #91. These will change after this PR is merged.

This closes #103.

### CODE REVIEW

**Comment by giwon2004** [comment]
The `typing` element `ItemsView` for `SubmittedExperimentInfo.items()` in `protocols.py` is supported since Python 3.9.

File: iquip/protocols.py [hunk 1]
```diff
@@ -15,3 +15,27 @@ class ExperimentInfo:
     """
     name: str
     arginfo: Dict[str, Any]
+
```
Comment (BECATRUE) [inline iquip/protocols.py:18]: Please add one more empty line!

File: iquip/protocols.py [hunk 2]
```diff
@@ -15,3 +15,27 @@ class ExperimentInfo:
...
+    """Information holder for submitted experiments.
+    
+    Fields:
+        rid: The run identifier value of the experiment
+        priority: The priority of the experiment
+        status: Current state of the experiment
+        pipeline: The pipeline of the experiment
+        expid: The overall information of the experiment, 
+          which is a dictionary that may include arguments, file, etc.
+        due_date: The due date of the experiment
```
Comment (BECATRUE) [inline iquip/protocols.py]: Please add a period(.) into all sentences.

File: iquip/protocols.py [hunk 3]
```diff
@@ -15,3 +15,27 @@ class ExperimentInfo:
...
+@dataclasses.dataclass
+class SubmittedExperimentInfo:
+    """Information holder for submitted experiments.
+    
+    Fields:
+        rid: The run identifier value of the experiment
+        priority: The priority of the experiment
+        status: Current state of the experiment
+        pipeline: The pipeline of the experiment
+        expid: The overall information of the experiment,
```
Comment (BECATRUE) [inline iquip/protocols.py:29]: As I know, `expid` contains the `priority`, `pipeline`, and `due_date`.
If so, are those fields unnecessary?

File: iquip/protocols.py [hunk 4]
```diff
@@ -15,3 +15,27 @@ class ExperimentInfo:
...
+    """
+    rid: int
+    priority: int = 0
+    status: str = ""
+    pipeline: str = ""
+    expid: Dict[str, Any] = {}
+    due_date: str = ""
+
+    def items(self) -> ItemsView[str, Any]:
+        """Returns the attributes of the experiment in the type of dict_items"""
```
Comment (BECATRUE) [inline iquip/protocols.py]: I think "in the type ..." is unnecessary because the type hinting indicates that!

File: iquip/apps/scheduler.py [hunk 5]
```diff
@@ -91,34 +87,27 @@ def updateInfo(self, info: Optional[ExperimentInfo] = None):
...
         """Extended.
         
         Args:
             info: The information of the experiment.
         """
         super().__init__(parent=parent)
         # widgets
-        self.nameLabel = QLabel(info.name, self)
-        labels = (QLabel(f"{key}: {value}", self) for key, value in info.arginfo.items())
+        labels = tuple(QLabel(f"{key}: {value}", self) for key, value in info.items())
```
Comment (BECATRUE) [inline iquip/apps/scheduler.py:108]: Is `tuple()` necessary?

**Review by BECATRUE — CHANGES_REQUESTED** [review]

**Comment by BECATRUE** [comment]
Also, could you add an example screenshot?

File: iquip/apps/scheduler.py (hunk 5 above)
Comment (giwon2004) [reply iquip/apps/scheduler.py:108]: If not, the parantheses are just an envelope that cases `generator`-typed value. Thus, the type of `labels` would be `<class 'generator'>`.

File: iquip/protocols.py (hunk 3 above)
Comment (BECATRUE) [reply iquip/protocols.py:29]: Oh, I'm really sorry. The `expid` doesn't contain those arguments.
Then, your implement looks right.

**Comment by giwon2004** [comment]
I think I need to rebase this branch to have unit tests also on the track, as unit tests are merged now by #89.

**Comment by giwon2004** [comment]
Commit [68e9164](https://github.com/snu-quiqcl/iquip/pull/104/commits/68e91648cdb190a76e8dc093d19fd5b25ce6cd3c) shows the difference between the original unit test (manually rebased by commit [9b9b9a6](https://github.com/snu-quiqcl/iquip/pull/104/commits/9b9b9a66551c3c3a4c66026a5ae1b36df3988a8e)) and the modified version.

**Review by kangz12345 — APPROVED** [review]
I approve this since I don't want to block the merge schedule.
If you need my opinions, please mention me.

**Comment by giwon2004** [comment]
> I approve this since I don't want to block the merge schedule. If you need my opinions, please mention me.

You can make a new change request on other PRs, as they cover almost all of what I have changed here.

**Review by BECATRUE — APPROVED** [review]
LGTM!

**Pull request merged**